datafusion/core/tests/sql/errors.rs

```diff
@@ -132,41 +132,3 @@ async fn invalid_qualified_table_references() -> Result<()> {
     }
     Ok(())
 }
-
-#[tokio::test]
```

Comment:
I would still like this test to be present (updated to call `SessionState::create_logical_plan`)

I think it is important to have this test because it communicates the need / intent for the "don't mutate session state" APIs